speasy/config/__init__.py

```diff
@@ -1,24 +1,50 @@
+"""
+speasy.config
+-------------
+
+Configuration module.
+"""
+
 import configparser, os
 import appdirs
 from ..common import mkdir
+from .exceptions import *
 
 _CONFIG_FNAME = str(appdirs.user_config_dir(appname="speasy", appauthor="LPP")) + "/config.ini"
 mkdir(os.path.dirname(_CONFIG_FNAME))
 _config = configparser.ConfigParser()
 _config.read(_CONFIG_FNAME)
 
 
+
 class ConfigEntry:
+    """Configuration entry class. Used to set and get configuration values.
+
+    :param key1: key 1
+    :type key1: str
+    :param key2: key 2
+    :type key2: str
+    :param default: default value
+    :type default: str
+    """
     def __init__(self, key1, key2, default=""):
         self.key1 = key1
         self.key2 = key2
         self.default = default
 
     def get(self):
+        """Get configuration entry value. If a default is not provided then raise :class:`~speasy.config.exceptions.UndefinedConfigEntry`.
+
+        :return: configuration value
+        :rtype: str
+        """
         if self.key1 in _config and self.key2 in _config[self.key1]:
             return _config[self.key1][self.key2]
         else:
-            return self.default
+            if len(self.default):
+                return self.default
+            # key is not defined, raise an UndefinedConfigEntry exception
+            raise UndefinedConfigEntry(key1=self.key1, key2=self.key2, default=self.default)
```

Comment:
This should not raise here, most of the time this is ok to have empty config entry.

Reply:
ok. I have reverted the code to its older version.